legend-engine-xts-data-space/legend-engine-xt-data-space-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/DataSpaceCompilerExtension.java

```diff
@@ -282,6 +291,79 @@ else if (dataSpace.supportInfo instanceof DataSpaceSupportCombinedInfo)
+                                            Root_meta_pure_metamodel_valuespecification_SimpleFunctionExpression_Impl fromFuncExpression = (Root_meta_pure_metamodel_valuespecification_SimpleFunctionExpression_Impl) fromFunc.get();
+
+                                            // only check mapping and runtime if using ->from() in the most basic way e.g. ->from(model::Mapping, model::Runtime)
+                                            ValueSpecification mappingInstance = fromFuncExpression._parametersValues().toList().get(1);
+                                            if (mappingInstance instanceof Root_meta_pure_metamodel_valuespecification_InstanceValue_Impl)
+                                            {
+                                                String executionContextKey = ((DataSpaceTemplateExecutable) executable).executionContextKey != null ? ((DataSpaceTemplateExecutable) executable).executionContextKey : dataSpace.defaultExecutionContext;
+                                                DataSpaceExecutionContext executionContext = dataSpace.executionContexts.stream().filter(ec -> ec.name.equals(executionContextKey)).collect(Collectors.toList()).get(0);
+                                                // check if mapping matches to what is used in execution key
+                                                Object mappingImpl = ((Root_meta_pure_metamodel_valuespecification_InstanceValue_Impl) mappingInstance)._values().toList().get(0);
+                                                if (mappingImpl instanceof Root_meta_pure_mapping_Mapping_Impl)
+                                                {
+                                                    String mappingPath = platform_pure_essential_meta_graph_elementToPath.Root_meta_pure_functions_meta_elementToPath_PackageableElement_1__String_1__String_1_((Root_meta_pure_mapping_Mapping_Impl) mappingImpl, "::", context.pureModel.getExecutionSupport());
+                                                    if (!mappingPath.equals(executionContext.mapping.path))
+                                                    {
+                                                        throw new EngineException("The mapping utilized in the function within the curated template query does not align with the mapping applied in the execution context `" + executionContext.name + "`.");
+                                                    }
+                                                }
+                                                // check if runtime matches to what is used in execution key
+                                                RichIterable<? extends org.finos.legend.pure.generated.Root_meta_core_runtime_Runtime> runtimes = org.finos.legend.pure.generated.core_pure_corefunctions_metaExtension.Root_meta_pure_functions_meta_extractRuntimesFromFunctionDefinition_FunctionDefinition_1__Runtime_MANY_(templateExecutableQuery, context.pureModel.getExecutionSupport());
```

Comment:
Is there a similar function for mappings ? 
If not can we please build one out so its consistent with how we grab mappings